app/controllers/katello/products_controller.rb

```diff
@@ -80,8 +79,8 @@ def exclude_rolling_kickstart_repos(repos)
       end
     end
 
-    def available_synced_repos(repos)
-      @product.repositories.each do |product_repo|
+    def available_synced_repos(repos, content_id)
+      @product.repositories.in_default_view.where(:content_id => content_id).each do |product_repo|
```

Comment:
Yeah it is pretty rare, but i don't think there's a reason to not use find_each. 👍